avro/profile.json

```diff
@@ -0,0 +1,118 @@
+{
+  "name":"profile",
+  "type":"record",
+  "namespace": "dto",
+  "doc":"Schema for cosmos api container 'profiles'",
+  "fields":[
+    {
+      "name": "id",
+      "type": "string",
+      "default": "undefined"
+    },
+    {
+      "name":"version",
+      "type":"int",
+      "default":0
+    },
+    {
+      "name": "timestamp",
+      "type": "long",
+      "logicalType": "timestamp-millis",
+      "default": 0
+    },
+    {
+      "name":"blockedInboxOrChannels",
+      "doc": "Contains blocked channels per each service. This info remains for backward compatibility only, it has been replaced with service-preferences data",
+      "type":{
+        "type": "map",
+        "values":{
+          "type": "array",
+          "items": {
+            "type":"enum",
+            "name":"BlockedInboxOrChannelEnum",
+            "symbols":[
+              "EMAIL",
+              "WEBHOOK",
+              "INBOX"
+            ]
+          }
+        }
+      },
+      "default" : {}
+    },
+    {
+      "name":"userPDVId",
+      "type": "string",
+      "default": "undefined"
+    },
+    {
+      "name": "isEmailEnabled",
+      "type": "boolean"
+    },
+    {
+      "name": "isEmailValidated",
+      "type": "boolean"
+    },
+    {
+      "name": "isInboxEnabled",
+      "type": "boolean"
+    },
+    {
+      "name": "isWebhookEnabled",
+      "type": "boolean"
+    },
+    {
+      "name":"lastAppVersion",
+      "type": "string",
+      "default": "UNKNOWN"
+    },
+    {
+      "name":"pushNotificationsContentType",
+      "type":{
+        "name":"PushNotificationsContentTypeEnum",
+        "type":"enum",
+        "symbols":[
+          "UNSET",
+          "FULL",
+          "ANONYMOUS"
+        ]
+      },
+      "default": "UNSET"
+    },
+    {
+      "name": "reminderStatus",
+      "type": {
+        "type":"enum",
+        "name":"ReminderStatus",
+        "symbols":[
+          "ENABLED",
+          "DISABLED",
+          "UNSET"
+        ]
+      },
+      "default": "UNSET"
+    },
+    {
+      "name":"servicePreferencesSettings",
+      "type": {
+        "name":"ServicesPreferencesMode",
+        "type": "record",
+        "fields":[
+          { 
+            "name":"mode",
+            "type":{
+              "type":"enum",
+              "name":"ServicesPreferencesModeEnum",
+              "symbols":["AUTO","MANUAL","LEGACY"]
+            }
+          },
+          {
+            "name":"version",
+            "type":"int",
+            "default": 0
```

Comment:
this default should be `-1` with legacy mode. But we can handle the default values on the read model, if we decode the value using del model type, all the `servicePreferencesSettings` will have the following value:
```
servicePreferencesSettings: withDefault(ServicePreferencesSettings, {
  mode: ServicesPreferencesModeEnum.LEGACY,
  version: PROFILE_SERVICE_PREFERENCES_SETTINGS_LEGACY_VERSION
})
```

Reply:
initially I put to separated types. if you think we can change the decoder behaviour, I'll leave this as `0` by default

Reply:
or removing the default?

Reply:
I'll add a comment too

Reply:
done in https://github.com/pagopa/io-functions-elt/pull/81/commits/2f624297a89404d70d65a11999e722f0332b8583